aws-event-bridge/src/test/scala/akka/stream/alpakka/aws/eventbridge/EventBridgePublishMockSpec.scala

```diff
@@ -0,0 +1,176 @@
+import akka.stream.scaladsl.{Keep, Sink}
+import akka.stream.testkit.scaladsl.TestSource
+import org.mockito.ArgumentMatchers.{any, eq => meq}
+import org.mockito.Mockito._
+import org.scalatest._
+import software.amazon.awssdk.services.eventbridge.model._
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class EventBridgePublishMockSpec extends FlatSpec with DefaultTestContext with MustMatchers {
+
+  private def entryDetail(detail: String): PutEventsRequestEntry =
+    PutEventsRequestEntry.builder().detail(detail).build()
+
+  private def requestDetail(detail: String): PutEventsRequest =
+    PutEventsRequest.builder().entries(entryDetail(detail)).build()
+
+  private def resultResponse(eventId: String): PutEventsResponse = {
+    val putResultEntry = PutEventsResultEntry.builder().eventId(eventId).build()
+    PutEventsResponse.builder().entries(putResultEntry).build()
+  }
+
+  it should "publish a single PublishRequest message to eventBridge" in {
+    val putRequestEntry = PutEventsRequestEntry.builder().detail("event-bridge-message").build()
+    val putRequest = requestDetail("event-bridge-message")
+    val putResult = resultResponse("event-id")
+
+    when(eventBridgeClient.putEvents(putRequest)).thenReturn(CompletableFuture.completedFuture(putResult))
+
+    val (probe, future) =
+      TestSource.probe[PutEventsRequest].via(EventBridgePublisher.publishFlow()).toMat(Sink.seq)(Keep.both).run()
+
+    probe.sendNext(PutEventsRequest.builder().entries(putRequestEntry).build()).sendComplete()
+
+    Await.result(future, 1.second) mustBe putResult :: Nil
+
+    verify(eventBridgeClient, times(1)).putEvents(meq(putRequest))
+  }
+
+  it should "publish multiple PutEventsRequest messages to eventbridge" in {
+    val putResultEntry = PutEventsResultEntry.builder().eventId("event-id").build()
+    val putResult = PutEventsResponse.builder().entries(putResultEntry).build()
+
+    when(eventBridgeClient.putEvents(any[PutEventsRequest]())).thenReturn(CompletableFuture.completedFuture(putResult))
+
+    val (probe, future) =
+      TestSource.probe[PutEventsRequest].via(EventBridgePublisher.publishFlow()).toMat(Sink.seq)(Keep.both).run()
+
+    probe
+      .sendNext(requestDetail("eb-message-1"))
+      .sendNext(requestDetail("eb-message-2"))
+      .sendNext(requestDetail("eb-message-3"))
+      .sendComplete()
+
+    Await.result(future, 1.second) mustBe putResult :: putResult :: putResult :: Nil
+
+    val expectedFirst = requestDetail("eb-message-1")
+    verify(eventBridgeClient, times(1)).putEvents(meq(expectedFirst))
+
+    val expectedSecond = requestDetail("eb-message-2")
+    verify(eventBridgeClient, times(1)).putEvents(meq(expectedSecond))
+
+    val expectedThird = requestDetail("eb-message-3")
+    verify(eventBridgeClient, times(1)).putEvents(meq(expectedThird))
+  }
+
+  it should "publish multiple PublishRequest messages to multiple eventbridge topics" in {
```

Comment:
It's not clear to me how this test is routing or asserting that messages are sent to _different_ topics.  It looks like the same impl as above.